src/replayq_registry.erl

```diff
@@ -0,0 +1,108 @@
+-module(replayq_registry).
+
+-behaviour(gen_server).
+
+%% API
+-export([
+    start_link/0,
+
+    register_committer/2,
+    deregister_committer/1
+]).
+
+%% `gen_server' API
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%%------------------------------------------------------------------------------
+%% Type declarations
+%%------------------------------------------------------------------------------
+
+%% call/cast/info events
+-record(register_committer, {dir :: filename:filename_all(), pid :: pid()}).
+-record(deregister_committer, {pid :: pid()}).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+register_committer(Dir, Pid) ->
+    gen_server:call(?MODULE, #register_committer{dir = Dir, pid = Pid}, infinity).
+
+deregister_committer(Pid) ->
+    gen_server:call(?MODULE, #deregister_committer{pid = Pid}, infinity).
+
+%%------------------------------------------------------------------------------
+%% `gen_server' API
+%%------------------------------------------------------------------------------
+
+init(_Opts) ->
+    process_flag(trap_exit, true),
+    State = #{committers => #{}},
+    {ok, State}.
+
+handle_call(#register_committer{dir = Dir, pid = Pid}, _From, State0) ->
+    %% Should we expand the directory path to avoid tricks with links and relative paths?
```

Comment:
good point.

Reply:
noticed that we should also normalize the dir (i.e.: don't treat an iolist, string and binary representations of the same path differently).  pushed a small fix for that